shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -770,14 +770,21 @@ public ExclusiveAppComponent<Activity> getExclusiveAppComponent() {
     return delegate;
   }
 
+  // We track the attached context so we can unregister component callbacks when detaching.
+  private Context attachedContext;
+
   @Override
   public void onAttach(@NonNull Context context) {
     super.onAttach(context);
-    delegate = new FlutterActivityAndFragmentDelegate(this);
+    if (delegate == null) {
+      delegate = new FlutterActivityAndFragmentDelegate(this);
+    }
```

Comment:
This seems strange - maybe we're constructing the test incorrectly?

I'm under the impression that onAttach should never get called twice before onDetach is called in a real case.

Reply:
Specifically here, I am passing in a mock delegate that is being blindly overwritten here. The delegate needs to be mocked as it requires a ton of setup to be able to successfully call delegate.onAttach(). This null check allows us to mock it.

Reply:
Can we instead have a delegate factory this class uses, perhaps only visible for testing?

Reply:
Changed to use a factory.